### PR TITLE
HDDS-8186. `ozone sh key put` does not work in single-datanode compose example

### PR DESCRIPTION
**Pull request by SaketaChalamchala**
## What changes were proposed in this pull request?

The [documentation](https://ozone.apache.org/docs/1.3.0/start/runningviadocker.html) has instructions to create a single DN node Ozone docker environment. But, key create commands fail because server-side default replication config is RATIS THREE.
We set client replication config to 1 in client configuration in the docker environments. Also, set the replication type to RATIS so that keys can be created with RATIS ONE replication type in a single DN setup. 

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-8186

## How was this patch tested?

Manual.
Follow [documentation](https://ozone.apache.org/docs/1.3.0/start/runningviadocker.html) to spin up a docker cluster.
Run
```
ozone sh volume create /vol1
ozone sh bucket create /vol1/bucket1
touch /tmp/myfile
ozone sh key put /vol1/bucket1/key1 /tmp/myfile 
```

### CODE REVIEW

**Review by adoroszlai — CHANGES_REQUESTED** [review]
Thanks @SaketaChalamchala for reporting this problem.

I think we should simply add the default replication to `docker-config`:

```
OZONE-SITE.XML_ozone.replication.type=RATIS
```

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/bucket/CreateBucketHandler.java [hunk 1]
```diff
@@ -109,8 +110,12 @@ public void execute(OzoneClient client, OzoneAddress address)
...
     }
 
-    replication.fromParams(getConf()).ifPresent(config ->
-        bb.setDefaultReplicationConfig(new DefaultReplicationConfig(config)));
+    ReplicationConfig replicationConfig =
+        replication.fromParamsOrConfig(getConf());
+    if (replicationConfig != null) {
+      bb.setDefaultReplicationConfig(
+          new DefaultReplicationConfig(replicationConfig));
+    }
```
[the remark this answers is not in the document]
Comment (SaketaChalamchala) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/shell/bucket/CreateBucketHandler.java]: Yes, you're right. This should be a simple docker compose change. I updated the patch. Thanks @adoroszlai !

**Comment by adoroszlai** [comment]
Thanks @SaketaChalamchala for updating the patch.  I think it does what is intended.  However, it uncovered a bug in Ozone CLI related to EC (HDDS-9035), which I'll need to fix before this can be merged.  I'll mark this as draft until then.

**Comment by adoroszlai** [comment]
@umamaheswararao please review

**Review by sadanand48 — COMMENTED** [review]
Thanks @SaketaChalamchala for making this change. This change would create a key with r.f=1 by default in `compose/ozone` env . Although if I intend to  test out the same  env with multiple DN's using `docker-compose --scale datanode=3` it still creates ONE replication keys. 

```bash
 ozone % docker-compose up --scale datanode=3 -d
bash-4.2$ ozone sh key put vol1/buck1/key1 README.md
bash-4.2$ ozone sh key info vol1/buck1/key1 | grep 'replicationFactor'
replicationFactor : ONE
```
An improvement here could be to read value of --scale and set the config. I'm not sure if this is possible but just a suggestion.

**Comment by SaketaChalamchala** [comment]
> An improvement here could be to read value of --scale and set the config. I'm not sure if this is possible but just a suggestion.

I don't think this can be done @sadanand48 . To use RATIS THREE you would have to do 
`OZONE_REPLICATION_FACTOR=3 docker-compose up --scale datanode=3 -d`

**Review by adoroszlai — APPROVED** [review]

**Pull request merged**